drivers/SmartThings/zwave-sensor/src/ezmultipli-multipurpose-sensor/init.lua

```diff
@@ -0,0 +1,129 @@
+-- Copyright 2022 SmartThings
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+
+local capabilities = require "st.capabilities"
+--- @type st.utils
+local utils = require "st.utils"
+--- @type st.zwave.CommandClass
+local cc = require "st.zwave.CommandClass"
+--- @type st.zwave.constants
+local constants = require "st.zwave.constants"
+--- @type st.zwave.CommandClass.Basic
+local Basic = (require "st.zwave.CommandClass.Basic")({version=1,strict=true})
+--- @type st.zwave.CommandClass.SwitchColor
+local SwitchColor = (require "st.zwave.CommandClass.SwitchColor")({version=3,strict=true})
+--- @type st.zwave.CommandClass.SwitchBinary
+local SwitchBinary = (require "st.zwave.CommandClass.SwitchBinary")({version=2,strict=true})
+
+local CAP_CACHE_KEY = "st.capabilities." .. capabilities.colorControl.ID
+
+local EZMULTIPLI_MULTIPURPOSE_SENSOR_FINGERPRINTS = {
+  { manufacturerId = 0x001E, productType = 0x0004, productId = 0x0001 }
+}
+
+local function can_handle_ezmultipli_multipurpose_sensor(opts, driver, device, ...)
+  for _, fingerprint in ipairs(EZMULTIPLI_MULTIPURPOSE_SENSOR_FINGERPRINTS) do
+    if device:id_match(fingerprint.manufacturerId, fingerprint.productType, fingerprint.productId) then
+      return true
+    end
+  end
+  return false
+end
+
+local function basic_report_handler(driver, device, cmd)
+  local event
```

Comment:
Can we do something like to avoid code duplication ?

local function basic_report_handler(driver, device, cmd)
  local event
  local value = (cmd.args.target_value !=nil) and cmd.args.target_value or cmd.args.value
     -- Target value is our best indicator of eventual state.
    -- If we see this, it should be considered authoritative.
    if cmd.args.target_value == SwitchBinary.value.OFF_DISABLE then
      event = capabilities.switch.switch.off()
      device:emit_event_for_endpoint(cmd.src_channel, capabilities.colorControl.hue(0))
      device:emit_event_for_endpoint(cmd.src_channel, capabilities.colorControl.saturation(0))
    else
      event = capabilities.switch.switch.on()
    end

  device:emit_event_for_endpoint(cmd.src_channel, event)
end

Reply:
I agree

Reply:
sure, why not